deps/amqp10_client/src/amqp10_msg.erl

```diff
@@ -270,14 +270,23 @@ new(DeliveryTag, Bin, Settled) when is_binary(Bin) ->
     Body = [#'v1_0.data'{content = Bin}],
     new(DeliveryTag, Body, Settled);
 new(DeliveryTag, Body, Settled) -> % TODO: constrain to amqp types
-    #amqp10_msg{
-       transfer = #'v1_0.transfer'{
-                     delivery_tag = {binary, DeliveryTag},
-                     settled = Settled,
-                     message_format = {uint, ?MESSAGE_FORMAT}},
-       %% This lib is safe by default.
-       header = #'v1_0.header'{durable = true},
-       body = Body}.
+    case is_amqp10_body(Body) orelse (not is_list(Body)) of
+        true ->
+            #amqp10_msg{
+               transfer = #'v1_0.transfer'{
+                             delivery_tag = {binary, DeliveryTag},
+                             settled = Settled,
+                             message_format = {uint, ?MESSAGE_FORMAT}},
+               %% This lib is safe by default.
+               header = #'v1_0.header'{durable = true},
+               body = Body};
+        false ->
+            Transfer = #'v1_0.transfer'{
+                          delivery_tag = {binary, DeliveryTag},
+                          settled = Settled,
+                          message_format = {uint, ?MESSAGE_FORMAT}},
+            from_amqp_records([Transfer | Body])
+    end.
```

Comment:
The function spec is now wrong: The function spec expects currently a body, but not other sections.

Reply:
The creation of the `'v1_0.transfer'` record got duplicated. Avoid code duplication.